### ISSUE TITLE
Possibility to run on Linux

### ISSUE THREAD

**Issue by DazDotOne** [LeeHolmes/watchysim#11]
Hi,

Firstly, as always, thanks for any and all work you've done on this. I'm a seasoned developer but very new to embedded systems and have recently acquired a watchy which I'm loving.

I'm looking at creating some faces (and possibly some drivers for different displays).

I'd love to be able to sim the watchy but I run Linux.

I was just wondering if you'd had any interactions with doing so and whether you had any pointers for where to start if not.

Thanks again

D.

**Comment by LeeHolmes** [comment]
Hey Daniel - right now, it adapts the Watchy SDK (which is pretty fully the Arduino SDK with some additions) and maps it all into a GDI application for Windows. But if you're eager, I think you could port it to run on Linux without a ton of effort. The WatchySim is essentially just a bunch of mocks.

For example, if your Watchy watch face calls "print(<some string>)", there is a mock of that in Watchy.cpp (DisplaySim::print) that calls an mocked Arduino API (drawFontBitmap(...)). That does a bunch of stuff that finally calls into a few GDI-native functions like "setPixel". Then all of this is hosted inside a Windows GDI application.

If you wanted to port this to Linux, most of the work would just be cloning the very simple user interface, wiring the menu items (like setting the time) to code that already exists, and porting the handful of functions that do the actual drawing to a display area (like drawing a pixel, drawing a rectangle, drawing a line, measuring a string, and maybe a few others).

The vast majority of the code is mocking the Watchy SDK and Arduino SDKs into those primitive functions and wouldn't need to change.

**Comment by DazDotOne** [comment]
Amazing! Thanks for the response Lee.

I'll look at having a play then. Didn't want to go down a rabbit hole if it was going to be a no go.

I'll let you know if I make any progress.

**Comment by LeeHolmes** [comment]
Would love to hear how it goes.

BTW, one thing that is easy to miss - it MUST be a C or C++-based UI framework that you use. There are lots of very capable UI frameworks on Linux for Python etc (i.e.: I made an oscilloscope app: 
LeeHolmes/nanoscope: Front end to convert your Arduino Nano into a fully featured Oscilloscope (github.com) 
<https://github.com/LeeHolmes/nanoscope>), but the point of WatchySim is to be able to accelerate Watchy development (which is Arduino C and C++).

I looked into Qt at one point, but it was a bloated mess. There are of course many others: C++ UI Libraries • memdump (philippegroarke.com) <https://philippegroarke.com/posts/2018/c++_ui_solutions/>

**Comment by DazDotOne** [comment]
Thanks for the heads up. 

Yeah I've been toying with the idea of making something in Qt for a while now but held off for similar reasons (and not having a decent endgame).

Guess this might be a good chance to get my hands dirty 😬

**Comment by hookdump** [comment]
Hey all! I just got my watchy today, and I didn't feel like restarting my computer and booting Windows, so I asked Claude Code to build a Linux port, and after a few minutes I got the basic functionality working (rendering the watchface).

I opened a PR in the main repo, just in case y'all are interested in merging this:
https://github.com/LeeHolmes/watchysim/pull/15

Alternatively, you can access it in my fork:
https://github.com/hookdump/watchysim-linux/tree/linux-port

Cheers!!

<img width="710" height="726" alt="Image" src="https://github.com/user-attachments/assets/a1548f63-3814-431a-9859-2855f7d024d5" />


<img width="701" height="719" alt="Image" src="https://github.com/user-attachments/assets/48f5463c-e6d5-40a8-b5f1-dbfbce584fc7" />

**Comment by LeeHolmes** [comment]
This is fantastic, @hookdump, thank you!

I will absolutely review and merge this once you think it meets your needs on Linux! A very useful part of the simulator on Windows is that you have menu items to test edge cases - things like changing the date, setting a step count to a 5-digit number, testing all the weather options, etc. It looks like you already have that on your mind based on [this comment](https://github.com/hookdump/watchysim-linux/blob/2f0e02d7d7fd380c3141466f8d828cc78632fa41/WatchySim/WatchySim_SDL2.cpp#L5) - want to let me know when you get there?

**Comment by hookdump** [comment]
Absolutely, will do! I'll comment in the PR and tag you once that's ready.

**Comment by hookdump** [comment]
Done!

<img width="1019" height="717" alt="Image" src="https://github.com/user-attachments/assets/a14c09d1-07aa-4981-a2f9-9e6bf84ebda6" />

I still want to do some more polishing before the PR is 100% ready for review (i.e. making sure all the example watchfaces can build properly -- most do, but I haven't tested some!). I'll ping you once that's ready. :)

**Comment by dsalt** [comment]
> Generated with Claude Code
LLM? Seriously? “Vibe coding” is an instant “no, not using, get thee behind me”. Shame since a Linux port would otherwise be a good thing.
